### PR TITLE
Url resolution & menu bugs

### PR DESCRIPTION
**Pull request by MaanooAk**
Added url resolution on paste (based on #10).

Fill free to change anything (names, filenames...), just wanted to add the basic functionality open to possible refactoring.

### CODE REVIEW

File: app/js/utils.js [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+function testUrl(text) {
+  return /^https?:\/\/.*$/.test(text)
+}
+
+/**
+ * Test if a given string represents a MIME that may contain json
+ */
+
+function testContentType(text) {
+  return /^(application|text)\/((x-)?json5?|(x-)?javascript|plain)$/.test(text)
```
Comment (MaanooAk) [inline app/js/utils.js]: All combinations of: application, text / json, json5, javascript, x-..., plain

**Comment by MaanooAk** [comment]
Resolved conflicts (was **really** hard for some reason)

**Comment by wellsjo** [comment]
@MaanooAk probably because I just introduced `eslint` 😬 sorry

**Comment by MaanooAk** [comment]
#41 should close after this.

**Comment by MaanooAk** [comment]
Last commits removes the `template[1].submenu[7].submenu = ...` with fixed indexes (which has already caused enough problems).

**Comment by MaanooAk** [comment]
Fixing the validation errors now...

**Comment by wellsjo** [comment]
Nice, thank you for that. I haven't gotten the url resolution regex to work, perhaps we could leverage another library that has solved this problem such as https://github.com/ogt/valid-url?

**Comment by MaanooAk** [comment]
Wait, you mean it doesn't work for you?
(note: it has to be one line starting with http:// or https://)
Try copy this `http://maanoo.com/projects/data.json` and git to the app select all -> paste.

**Comment by wellsjo** [comment]
I see, yeah that was the issue. Thanks for cleaning up the validation errors! I'm down to merge this for now. We can improve it over time.

**Pull request merged**